### DISCUSSION TITLE
A Question

### DISCUSSION THREAD

**Discussion by SAGEof6iixPATHS** [hahwul/dalfox#190] [Q&A]
What is the best method to use dalfox??
I've been using gau, waybackurls, paramspider to find urls and sometime used gf tool too...
But recently i came across a Tool named Quickxss, it filters urls completely...(You should try that, maybe you get a new idea from that)
Lets get back to the question...
What method do you suggest that will have more chances of finding bugs...

I run this tool and work on other bugs while this is going on, but never found anything (yet):/

And have you considered trying out sqli?? Like dalfox is more focused on xss

**Comment by hahwul** [comment]
@SAGEof6iixPATHS 
Hi! Sorry for the late response 😭
I think the important part is param mining. Known pages of commonly known services are likely to have already been scanned with automation tools. Finding a new vector here is an unknown page or parameter. 

Dalfox basically performs parameter mining together, but through a custom word list, mining can be performed based on wider data.

```
$ dalfox url {TARGET} -W /wordlists/big.txt
```

**Comment by hahwul** [comment]
@SAGEof6iixPATHS 
And, Testing massive URLs in pipe/file mode often results in good results, but using them together during manual testing, such as burpsuite/zap, is also good. If you read raw-http-request and proceed with the scan, it is easy to process authentication and headers, so I personally prefer it.

```
$ dalfox file request.txt --rawdata
```

However, since this process is difficult to run on command-line every time, it is sometimes defined and used to run manual in "burp or zap's run applications". Small tip, The company I work for makes it an burp extension and uses it!

* https://www.hahwul.com/2019/12/29/run-other-application-on-burp-suiteburp/
* https://www.hahwul.com/2019/07/01/easy-security-testing-with-applications-bridge-in-zap/
